app/models/support_interface/vendor_api_requests_filter.rb

```diff
@@ -0,0 +1,58 @@
+module SupportInterface
+  class VendorAPIRequestsFilter
+    attr_reader :applied_filters
+
+    def initialize(params:)
+      @applied_filters = params
+    end
+
+    def filters
+      @filters ||= [free_text] + [status_code] + [provider]
+    end
+
+  private
+
+    def free_text
+      {
+        type: :search,
+        heading: 'Search',
+        value: applied_filters[:q],
+        name: 'q',
+      }
+    end
+
+    def status_code
+      options = VendorAPIRequest.distinct(:status_code).pluck(:status_code).map do |status_code|
+        {
+          value: status_code,
+          label: status_code,
+          checked: applied_filters[:status_code]&.include?(status_code),
+        }
+      end
+
+      {
+        type: :checkboxes,
+        heading: 'Status code',
+        name: 'status_code',
+        options: options,
+      }
+    end
+
+    def provider
+      options = VendorAPIRequest.includes(:provider).distinct(:provider_id).pluck(:provider_id).compact.map do |provider_id|
+        {
+          value: provider_id,
+          label: Provider.find(provider_id).name,
```

Comment:
This looks like an N+1. We should be able to get all providers in one request I think.

Reply:
:+1: Strange this didn't get picked up by the lazy-loading linter, anyway I've updated to https://github.com/DFE-Digital/apply-for-teacher-training/pull/3009/files#diff-2c9f873e52850265633344b9c21301c8R42

Reply:
Yeah, I don't think it works very well.